### DISCUSSION TITLE
Console Warning, [VueFire SSR]: Could not get the path of the data source

### DISCUSSION THREAD

**Discussion by asifaly** [vuejs/vuefire#1277] [Questions and Help] [answered]
I just started re-factoring my application by introducing vuefire today. Everything works like a charm. I am facing the following warning in my console. I am just using default vue3 app with TS spin up by Vite. 

![image](https://user-images.githubusercontent.com/6282968/212694862-110d2ea3-0160-4a2a-be84-ccdafc2f489e.png)

The trace points to this file.... with the following code

```
<script setup lang="ts">
import NewQuizCard from "@/components/dashboard/NewQuizCard.vue";
import NoQuizIntro from "@/components/dashboard/NoQuizIntro.vue";
import QuizList from "@/components/dashboard/QuizList.vue";
import WelcomeMessage from "@/components/dashboard/WelcomeMessage.vue";
import type { IQuiz } from "@/interfaces";
import { collection, query, where } from "firebase/firestore";
import { useCollection, useCurrentUser, useFirestore } from "vuefire";

const db = useFirestore();
const user = useCurrentUser();

const quizzes = useCollection<IQuiz>(
  query(collection(db, "quizes"), where("userid", "==", user.value?.uid))
);
</script>
```

I assume it is because of `user.value?` being empty on page load.  I tried modifying it to

```
const quizzes = user.value
  ? useCollection<IQuiz>(
      query(collection(db, "quizes"), where("userid", "==", user.value.uid))
    )
  : [];
```

But the warning still appears... any suggestions would be appreciated.

**Comment by posva** [comment, marked as answer]
You need to ssr key your query:

```ts
useCollection(..., { ssrKey: 'unique ssr key' })
```

**Reply by asifaly** [reply]
Thanks @posva  it does make the warning go away. I did see this in the [docs](https://vuefire.vuejs.org/guide/ssr.html#manual-ssr-keys). I do not have SSR enabled in the application and this is still required  ?

**Comment by m2xilon** [comment]
Also getting this warning without the application doing any SSR

**Reply by posva** [reply]
Will be fixed next. You can ignore the warning or provide a key at the moment 🙂
